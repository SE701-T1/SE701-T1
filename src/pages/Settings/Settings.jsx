/* eslint-disable react/jsx-boolean-value */
import React, { useEffect, useState } from 'react';
import ToggleButton from '@mui/material/ToggleButton';
import ToggleButtonGroup from '@mui/material/ToggleButtonGroup';
import { Typography } from '@mui/material';
import styles from './Settings.module.css';
import { getSelf, updateUser } from '../../api/UserAPI';

/**
 * The settings page displays a toggle button that the user can click on to enable or disable pairing.
 * If the off option is selected, then the value of pairingEnabled is set to false and it is set to
 * true otherwise.
 *
 * The component populates the initial value of the component based on the response received from the
 * request using the getSelf() function call.
 *
 */

export default function Settings() {
  const [pairingEnabled, setPairingEnabled] = useState();

  useEffect(async () => {
    const userInfo = await getSelf();
    setPairingEnabled(userInfo.pairingEnabled);
  }, []);

  const handleChange = async (event, pairingStatus) => {
    if (pairingStatus !== null) {
      setPairingEnabled(pairingStatus);
      await updateUser(pairingStatus);
    }
  };

  return (
    <>
      <Typography sx={{ marginRight: '10px', display: 'inline' }} variant="body1">
        Pairing Mode:
      </Typography>
      <ToggleButtonGroup
        className={`${styles.container}`}
        value={pairingEnabled}
        exclusive
        onChange={handleChange}
      >
        <ToggleButton value={true}>On</ToggleButton>
        <ToggleButton value={false}>Off</ToggleButton>
      </ToggleButtonGroup>
    </>
  );
}
