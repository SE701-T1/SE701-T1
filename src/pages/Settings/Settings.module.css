.pairing {
  display: inline;
  padding-top: 11px;
  margin-right: 11px;
}
