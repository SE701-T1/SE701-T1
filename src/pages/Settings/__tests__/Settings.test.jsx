import React from 'react';
import renderer from 'react-test-renderer';
import Settings from '../Settings';

describe('Render Login Button', () => {
  test('should display the Toggle text', () => {
    const tree = renderer.create(<Settings />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
