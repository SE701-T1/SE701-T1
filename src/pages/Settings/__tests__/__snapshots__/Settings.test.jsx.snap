// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Login Button should display the Toggle text 1`] = `
Array [
  <p
    className="MuiTypography-root MuiTypography-body1 css-1m3pcfa-MuiTypography-root"
  >
    Pairing Mode:
  </p>,
  <div
    className="MuiToggleButtonGroup-root container css-1gjgmky-MuiToggleButtonGroup-root"
    role="group"
  >
    <button
      aria-pressed={false}
      className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-ueukts-MuiButtonBase-root-MuiToggleButton-root"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
      value={true}
    >
      On
    </button>
    <button
      aria-pressed={false}
      className="MuiButtonBase-root MuiToggleButton-root MuiToggleButton-sizeMedium MuiToggleButton-standard MuiToggleButtonGroup-grouped MuiToggleButtonGroup-groupedHorizontal css-ueukts-MuiButtonBase-root-MuiToggleButton-root"
      disabled={false}
      onBlur={[Function]}
      onChange={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
      value={false}
    >
      Off
    </button>
  </div>,
]
`;
